zerver/actions/realm_settings.py

```diff
@@ -527,3 +546,10 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def delete_deactivated_realms() -> None:
+    deferred_work_event = {
+        "type": "scrub_deactivated_realms",
+    }
+    queue_json_publish("deferred_work", deferred_work_event)
```

Comment:
As I note below, I think this function should take a `realm: Realm` and enqueue an event for just one `realm_id`.  The reason for this is assume that 20 realms all deactivate at once, with immediate deletion -- that will end up with 20 "delete all expired realms" events, which is _safe_ (as long as we only have one queue worker) but some unnecessary queries.

We should also name this function `scrub_deactivated_realm`, for accuracy.

Reply:
Should we add a check whether realm should be scrubbed now or not inside the function like this 
```
if realm.scheduled_deletion_date is None or realm.scheduled_deletion_date > timezone_now():
    return
```
I was adding this so that we can write test for this function or should I write test for command in `test_management_commands`.

Reply:
We _can_ do that here, but correctness only says that we need to do it when we're in the queue worker, right before we're doing the work.  If we check it here, any number of things could happen between now, when we enqueue the work, and some arbitrary amount of time later when the worker actually gets to it.